addons/print/models/print_printer.py

```diff
@@ -100,7 +107,14 @@ def set_default_printer(self):
             self.env.ref("print.default_printer").is_default = True
 
     def printers(self, report_type=None, raise_if_not_found=False):
-        """Determine printers to use"""
+        """
+        Purpose: Determine printers to use for a given report type. If no printers are found
```

Comment:
nit: remove `Purpose` as it is already implied by writing a doc string. Also this method doesn't return a print.printer(). But also I don't think it is necessary to include the return type in the docstring as we don't tend to do this elsewhere.

It would be nice to have some clearer guidelines for docstrings though so we can standardise them everywhere :)

Reply:
Yeah I think clearer guidelines are needed for sure. I want to keep the return type just because it makes it clear which data type is being returned, I think if we were working with a statically typed language there would be no need (because we would already specify it) but because python is dynamically typed it can take on any type, so best to keep track of what we think it is. I know there was a story recently where a method was returning a different data type to what we expected so I think it might be better to keep it in. 

Personally I like having the "Purpose" as it makes the doc-string clear and differentiates that between the returns :smile:

Reply:
@ws-mps do we have any guidelines surrounding how docstrings should be written?

Reply:
I think having the return is fine, it's actually a requirement to have one for some docstring standards/formats. I do agree with Caleb on the purpose though, not really necessary to have to say that you are explaining the purpose since the docstring is documentation of a function.

Reply:
> @ws-mps do we have any guidelines surrounding how docstrings should be written?

We need to adopt a standard, we haven't done it yet (I believe). I have done some research on this which would be helpful to decide which one to use. Search this page on the wiki (StoryX1871)

Reply:
So I have removed the purpose but I want to keep the returns :smile: